src/goto-symex/slice_by_trace.cpp

```diff
@@ -79,6 +79,11 @@ void symex_slice_by_tracet::slice_by_trace(
   {
     exprt g_copy(*i);
 
+    DATA_INVARIANT(
+      g_copy.id() == ID_symbol || g_copy.id() == ID_not ||
+        g_copy.id() == ID_and || g_copy.id() == ID_constant,
+      "guards should only be and, symbol, constant, or `not'");
```

Comment:
Either put quote marks around all the items, or none - at the moment you only put quote marks around 'not'